Network/Socket/Handle.hs

```diff
@@ -17,7 +18,7 @@ import Network.Socket.Types
 
 socketToHandle :: Socket -> IOMode -> IO Handle
 socketToHandle s mode = do
-    let fd = fromIntegral $ fdSocket s
+    fd <- fromIntegral <$> fdSocket s
```

Comment:
This function should invalidate the socket's file descriptor making `shoud avoid` behaviour impossible.  Basically, emulate `close`, but without closing the socket's file descriptor.  Or if you prefer (where available) `dup(2)` the socket's file descriptor, and close the socket.

Reply:
Actually, replace that `should` with a MUST. Given that Sockets (in this branch of the code) have finalizers that automatically close them, and doing so would then break the Handle, and trigger a second close outside the IORef guard in sockets, this is NOT optional.  You must either invalidate the socket, or at least `dup` the file descriptor and pass that to the handle, but invalidation is I think more appropriate here, since continued I/O via the socket is not supported or expected.

Reply:
Thanks.
And done in 7e6eea86ae0fb1aece8b77be6646b133b6f18cc1